### PR TITLE
feat: Migrate Registry to spell

### PR DESCRIPTION
**Pull request by gurinderu**

### CODE REVIEW

File: aqua/package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
   },
   "homepage": "https://github.com/fluencelabs/registry",
   "devDependencies": {
-    "@fluencelabs/aqua": "0.10.3"
+    "@fluencelabs/aqua": "0.10.3",
```
Comment (folex) [inline aqua/package.json:33]: why do you need aqua dependency?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -67,7 +67,7 @@ jobs:
       - name: Setup node with self-hosted registry
         uses: actions/setup-node@v3
         with:
-          node-version: "18"
+          node-version: "16"
```
Comment (folex) [inline .github/workflows/tests.yml:70]: why? i thought flox is moving to 18 node

could you pleass put up a comment, motivating the version?

File: distro/src/lib.rs [hunk 3]
```diff
@@ -26,28 +22,31 @@ pub fn modules() -> std::collections::HashMap<&'static str, &'static [u8]> {
...
 }
 
-pub struct Script {
-    pub name: &'static str,
+pub struct DistrSpell {
+    /// AIR script of the spell
     pub air: &'static str,
-    pub period_sec: u32,
+    /// Initial key-value records for spells KV storage
+    pub kv: HashMap<&'static str, JValue>,
```
Comment (folex) [inline distro/src/lib.rs:29]: that field is usually called init_data

File: spell/spell.aqua [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        now <- PeerTimeTrunc.timestamp_sec()
+        last_clear_expired <- Spell.get_u32("clear_expired")
+        run_clean_expired <- or(not(last_clear_expired.success),  Compare.gte(now - last_clear_expired.num, config.expired_interval)) 
+        if run_clean_expired == true:
+            log("Running clear_expired job")
+            clear_expired()
+            Spell.set_u32("clear_expired", now)
+
+        last_renew_interval <- Spell.get_u32("renew_interval")
+        run_renew = or(not(last_renew_interval.success),  Compare.gte(now - last_renew_interval.num, config.renew_interval))
```
Comment (folex) [inline spell/spell.aqua:70]: can you DRY it to a separate function?

File: .github/workflows/tests.yml (hunk 2 above)
Comment (gurinderu) [reply .github/workflows/tests.yml:70]: cli wasn't released for the node 18 and I broke everything

File: spell/spell.aqua (hunk 4 above)
Comment (gurinderu) [reply spell/spell.aqua]: Fixed

File: aqua/package.json (hunk 1 above)
Comment (gurinderu) [reply aqua/package.json]: Droped

File: distro/src/lib.rs [hunk 5]
```diff
@@ -26,28 +22,31 @@ pub fn modules() -> std::collections::HashMap<&'static str, &'static [u8]> {
...
-            name: "registry-renew",
-            air: RENEW_AIR,
-            period_sec: RENEW_PERIOD_SEC,
-        },
-        Script {
-            name: "registry-replicate",
-            air: REPLICATE_AIR,
-            period_sec: REPLICATE_PERIOD_SEC,
+
+/// Decider's configuration needed for the correct decider start-up
```
Comment (kmd-fl) [inline distro/src/lib.rs:33]: nope

Comment (gurinderu) [reply distro/src/lib.rs]: fixed

**Review by kmd-fl — APPROVED** [review]

**Review by folex — APPROVED** [review]

**Pull request merged**